search_frontend/js/search.js

```diff
@@ -30,20 +31,50 @@
           'c':''
       };
 
-      this._nodeSearchData = ko.observableArray([]);
+      this._nodeSearchData = ko.observableArray();
+      this.structureData = ko.observableArray()
       this._page = ko.observable(0);
       this._total = ko.observable(0);
       this._activateTab = ko.observable(false);
     }
-
-    /** @todo fetch from api */
+        /* @todo find what type of node is the current tab*/
+    findStructure() {

```

Comment:
It seems what is being done here only need to be done once. The constructor is a good place this.